5 d2h style guide

vigneshwaran-selvaraj opened the pull request:
Closes #5

vigneshwaran-selvaraj committed:
refactor: [#5] Fixes the checkbox styling.

Vigneshwaran1810 committed:
feat: [#5] D2H style guide for base.

Vigneshwaran1810 committed:
feat: [#5] D2H style guide for button.

Vigneshwaran1810 committed:
feat: [#5] D2H style guide for checkbox.

Vigneshwaran1810 committed:
feat: [#5] D2H new component alert and confirm dialogs.

Vigneshwaran1810 committed:
feat: [#5] D2H styleguide for messages.

Vigneshwaran1810 committed:
feat: [#5] D2H styleguide for ip-radio.

Vigneshwaran1810 committed:
feat: [#5] D2H styleguide for ip-select.

Vigneshwaran1810 committed:
feat: [#5] D2H styleguide for textbox and textarea.

mageshravi commented on static/css/src/_base.scss:
```diff
@@ -121,3 +122,73 @@ a {
+    color: #97AAB4;
+    font-style: italic;
+
+    &::before {
+        content: "";
+
+        display: block;
+        position: absolute;
+        top: 0;
+        left: 0;
+        width: 4px;
+        height: 100%;
+        background-color: #A7B7C0;
+    }
```
Can this not be replaced with,

```css
margin-block-start: 4px solid #A7B7C0;
```

mageshravi commented on static/css/src/_base.scss:
```diff
@@ -121,3 +122,73 @@ a {
+ol {
+    margin: 0;
+    padding: 0;
+}
```
Missing the spacing between `<li>` elements.

> List spacing: 6px

![image](https://github.com/user-attachments/assets/6455c37a-a96e-4440-b423-fabc9899959f)

mageshravi commented on static/css/src/_base.scss:
```diff
@@ -121,3 +122,73 @@ a {
+ul {
+    margin: 0;
+    padding: 0;
+}
```
Missing the spacing between `<li>` elements. See my previous comment.

mageshravi commented on static/css/src/_m-alert.scss:
```diff
@@ -0,0 +1,70 @@
+    </div>
+</div>
+
+Styleguide Dialog.alert
```
Rename to `Dialog.Alert`

mageshravi commented on static/css/src/_m-confirm.scss:
```diff
@@ -0,0 +1,74 @@
+    <div>
+</div>
+
+Styleguide Dialog.confirm
```
Rename to `Dialog.Confirm`.

mageshravi commented on static/css/src/_m-ip-select.scss:
```diff
@@ -121,17 +124,17 @@ Styleguide SelectMenu.UIStates
             width: var(--icon-size);
             height: var(--icon-size);
-            background-image: url('../../images/sprite-ip-ctrls.svg');
+            background-image: url('../../images/icons-sprite.svg');
```
Asset URLs in Django repo must start with `/static/`.

mageshravi reviewed (changes requested):
📝 Changes required. See my comments.

Vigneshwaran1810 committed:
Refactor: [#5] STyling changes in Base.

- Styling changes made in blockquote.
- Adds the spacing between <li> elements.

Vigneshwaran1810 committed:
Refactor: [#5] Renames For KSS

Vigneshwaran1810 committed:
refactor: [#5] Changed the Asset URLs starting from `/static/`.

mageshravi committed:
feat: [#5] bash script

- Adds a bash script to build and copy styleguide assets into the keyframes-styleguide repo.

vigneshwaran-selvaraj committed:
feat: [#5] Adds the background color for the body.

vigneshwaran-selvaraj committed:
refactor: [#5] Styling changes in the btn.

- Fixes the padding and border.
- Removes the unwanted KSS and CSS.

vigneshwaran-selvaraj committed:
refactor: [#5] Remove sthe border in input checked for checkbox and radio.

vigneshwaran-selvaraj committed:
refactor: [#5] Remove sthe unwanted KSS and CSS.

vigneshwaran-selvaraj committed:
refactor: [#5] Centrises the spinner in the textbox and textarea.

vigneshwaran-selvaraj committed:
feat: [#5] Adds KSS for Non dismissible messages.

vigneshwaran-selvaraj committed:
refactor: [#5] Adds the default styling upon disabled btn hover and active.

mageshravi committed:
KSS config

- adds homepage.md
- adds bun.lockb
- updates git repo url and name (in package.json)

mageshravi committed:
Merge branch '5-d2h-style-guide' of github.com:Webinative/keyframes into 5-d2h-style-guide

mageshravi committed:
[#5] Update copy to styleguide script

- automates replacing /static/ with /keyframes-styleguide/static/ in the main.css file.

vigneshwaran-selvaraj committed:
refactor: [#5] Fixes the ip-select border.

vigneshwaran-selvaraj committed:
refactor: [#5] Fixes the cursor on hover state.

mageshravi commented on static/css/src/_base.scss:
```diff
@@ -121,3 +123,69 @@ a {
+ul {
+    margin: 0;
+    padding: 0;
+}
```
Why duplicate the css rules?

```css
/* Keep the KSS declarations for UL and OL as is. */

ul, ol {
    margin: 0;
    padding: 0;

    li {
        margin-block-end: 6px;
    }
}
```

mageshravi commented on static/css/src/_m-alert.scss:
```diff
@@ -0,0 +1,70 @@
+
+.m-alert {
+    --overlay-zindex: 1;
+    --window-zindex: 10;
```
Have the higher z-index value on top.

```css
--window-zindex: 10;
--overlay-zindex: 1;
```

mageshravi commented on static/css/src/_m-alert.scss:
```diff
@@ -0,0 +1,70 @@
+        box-shadow:
+        0px 2px 4px 0px #00000033,
+        0px 4px 6px 2px #0000001F,
+        0px 6px 8px 4px #0000000F;
```
Is the indentation correct (as seen in VS Code)?

mageshravi commented on static/css/src/_m-confirm.scss:
```diff
@@ -0,0 +1,74 @@
+
+.m-confirm {
+    --overlay-zindex: 1;
+    --window-zindex: 10;
```
Reorder z-index values.

mageshravi commented on static/css/src/_m-confirm.scss:
```diff
@@ -0,0 +1,74 @@
+        box-shadow:
+        0px 2px 4px 0px #00000033,
+        0px 4px 6px 2px #0000001F,
+        0px 6px 8px 4px #0000000F;
```
Check indentation.

mageshravi commented on static/css/src/_m-messages.scss:
```diff
@@ -80,7 +84,103 @@ Styleguide Messages.Levels
+    </li>
+</ul>
+<ul class="m-messages">
+    <li class="m-messages__item m-messages__item--error">
```
Could have used `{{ modifier_class }}` instead of hard-coding.

mageshravi commented on static/css/src/_m-ip-checkbox.scss:
```diff
@@ -77,14 +78,31 @@ Styleguide Checkbox.UIStates
+            --sprite-icon-size: 24px;
+
+            position: relative;
+            background: #098CEA;
```
Suggest using a new CSS variable `--primary-color: #098CEA` in `_base.scss`

mageshravi commented on static/css/src/_m-ip-checkbox.scss:
```diff
@@ -77,14 +78,31 @@ Styleguide Checkbox.UIStates
+        &:hover::before,
         &:focus::before {
-            background-position-x: calc(-2 * var(--sprite-icon-size));
+            background: linear-gradient(180deg, #60B9F8 0%, #098CEA 100%);
```
Suggest using a new CSS variable `--primary-gradient: linear-gradient(180deg, #60B9F8 0%, #098CEA 100%);` in `_base.scss`

mageshravi commented on static/css/src/_m-ip-radio.scss:
```diff
@@ -75,11 +78,28 @@ Styleguide Radio.UIStates
-            background-position-x: calc(-2 * var(--sprite-icon-size));
+            position: relative;
+            border: none;
+            background: #098CEA;
```
Use a CSS variable. See my comment in `ip-checkbox.scss` file.

mageshravi commented on static/css/src/_m-ip-radio.scss:
```diff
@@ -75,11 +78,28 @@ Styleguide Radio.UIStates
 
         &:focus::before {
-            background-position-x: calc(-2 * var(--sprite-icon-size));
+            background: linear-gradient(180deg, #60B9F8 0%, #098CEA 100%);
```
Use a CSS variable. See my comment in `ip-checkbox.scss` file.

mageshravi commented on static/css/src/_m-ip-select.scss:
```diff
@@ -1,7 +1,7 @@
 Select Menu
 
-Select-menu with searchable options.
+Select-menu.
```
Replace with "Standard dropdown-menu."

mageshravi commented on static/css/src/_m-ip-select.scss:
```diff
@@ -46,8 +45,8 @@ Styleguide SelectMenu.UIStates
-    --opt-txt--hover: inherit;
-    --opt-bg--hover: #d9d9d9;
+    --opt-txt--hover: #fff;
+    --opt-bg--hover: linear-gradient(180deg, #60B9F8 0%, #098CEA 100%);
```
Make use of the `--primary-gradient` variable.

mageshravi commented on static/css/src/_m-ip-select.scss:
```diff
@@ -219,89 +149,27 @@ Styleguide SelectMenu.UIStates
-            color: var(--opt-txt--hover);
-            background-color: var(--opt-bg--hover);
+        &:nth-child(even) {
+            background-color: #E6EEF1;
```
Use a local CSS variable `alt-row-bg`

mageshravi commented on static/css/src/_m-ip-textarea.scss:
```diff
@@ -0,0 +1,134 @@
+        &.pseudo-class-focus {
+            border-color: var(--ip-border--focus);
+            background-color: var(--ip-bg--focus);
+            box-shadow: 0px 3px 2px 0px #00000040 inset;
```
Suggest using the `rgba` format instead of hexcode.

mageshravi commented on static/css/src/_m-ip-textarea.scss:
```diff
@@ -0,0 +1,134 @@
+        }
+    }
+
+    &.has-error &__error {
+        display: block;
+    }
+    }
```
Check the indentation.

mageshravi reviewed (changes requested):
📝 Multiple changes required.

vigneshwaran-selvaraj committed:
refactor: [#5] Removes the duplicated CSS.

vigneshwaran-selvaraj committed:
refactor: [#5] Used CSS variable insted of repeating colors.

vigneshwaran-selvaraj committed:
refactor: [#5] Fixes intendations.

vigneshwaran-selvaraj committed:
refactor: [#5] Changed the color code to RGBA.

vigneshwaran-selvaraj committed:
refactor: [#5] Replaced with  {{ modifier_class }} instead of hard-coding.

mageshravi commented on static/css/src/_m-ip-select.scss:
```diff
@@ -155,7 +155,9 @@ Styleguide SelectMenu.UIStates
 
         &:nth-child(even) {
-            background-color: #E6EEF1;
+            --alt-row-bg: #E6EEF1;
```
This can be higher up in the `block` scope. In the same level as the `--opt-bg--hover` variable.

mageshravi commented on static/css/src/_base.scss:
```diff
@@ -1,28 +1,31 @@
     --ip-border--focus: #898989;
     --ip-border--disabled: hsla(0, 0%, 85%, 0.5);
-    --ip-bg: #fafafa;
+    --ip-bg: #fff;
```
Per the design, the input element's default background is transparent (same as body-bg). Is that not the case?

mageshravi commented on static/css/src/_base.scss:
```diff
@@ -1,28 +1,31 @@
+    --btn-txt: #5E7581;
+    --btn-txt--hover: #5E7581;
+    --btn-txt--active: #5E7581;
+    --btn-bg: #fff;
```
Per the design, button background is transparent. Why is this white?

mageshravi commented on static/css/src/_m-ip-checkbox.scss:
```diff
@@ -29,28 +29,29 @@ Styleguide Checkbox.UIStates
-            cursor: pointer;
+            margin-inline-end: 10px;
+            border: 1px solid var(--ip-border);
+            background: transparent;
```
Should be using the `--ip-bg` variable.

mageshravi commented on static/css/src/_m-ip-checkbox.scss:
```diff
@@ -29,28 +29,29 @@ Styleguide Checkbox.UIStates
         &:hover::before,
         &.pseudo-class-hover::before {
-            background-position-x: calc(-1 * var(--sprite-icon-size));
+            background: #fff;
```
Should be using the `--ip-bg--hover` variable.

mageshravi commented on static/css/src/_m-ip-checkbox.scss:
```diff
@@ -29,28 +29,29 @@ Styleguide Checkbox.UIStates
         &:focus::before,
         &.pseudo-class-focus::before {
-            background-position-x: calc(-1 * var(--sprite-icon-size));
+            background: #fff;
```
Fix this and all other instances of `input` elements' background.

mageshravi reviewed (changes requested):
Changes required. CSS variables declaration and usage needs to improve.

vigneshwaran-selvaraj committed:
refactor: [#5] Fixes the useage of css varibales

vigneshwaran-selvaraj committed:
feat: [#5] D2H styleguide pagination.

vigneshwaran-selvaraj committed:
feat: [#5] D2H style guide pills.

vigneshwaran-selvaraj committed:
refactor: [#5] Fixes the KSS naming,

vigneshwaran-selvaraj committed:
refactor: [#5] Fixes lcode formatting using prettier formatter.

mageshravi reviewed (approved):
Changes looking good.
